app/models/concerns/host_info_extensions.rb

```diff
@@ -23,11 +23,11 @@ module HostInfoExtensions
   # This is the base method that gathers information from all providers.
   def info
     info_hash = {}
+    info_hash['parameters'] = {}
 
     HostInfo.providers.each do |provider_class|
       provider = provider_class.new(self)
-      info = provider.host_info
-      info_hash.deep_merge! info if info
+      provider.host_info!(info_hash)
```

Comment:
What about info providers not in core?

Reply:
I did only find katello. My plan is to file PR.

Reply:
Oh Ansible has that too.

Reply:
I can deprecate the old method too.

Reply:
we also use the info provider API in (internal) plugins.

Reply:
Implemented a detection and deprecated warning.